docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md

```diff
@@ -0,0 +1,177 @@
+
+- [Flux](https://toolkit.fluxcd.io/guides/mozilla-sops/#azure) has better support for SOPS with cluster-side decryption.
+- An added layer of security as the private key used for decryption is protected in an external key vault.
+- Needs plugin to work with Helm([Helm Secrets](https://github.com/jkroepke/helm-secrets)) and Kustomization ([KSOPS](https://github.com/viaduct-ai/kustomize-sops))([kustomize-sopssecretgenerator](https://github.com/goabout/kustomize-sopssecretgenerator))
+- Does not scale with larger teams as each developer has to encrypt the secrets
+- The public key is sufficient for creating brand new files. The secret key is required for decrypting and editing existing files because SOPS computes a MAC on all values.  When using the public key solely to add or remove a field, the whole file should be deleted and recreated.
+
+## 2. Reference to secrets stored in an external key vault (recommended)
+
+This approach relies on a key management system like [Azure Key Vault](https://learn.microsoft.com/en-us/azure/key-vault/general/overview) to hold the secrets and the git manifest in the repositories has reference to the key vault secrets. Developers do not perform any cryptographic operations with files in repositories. Kubernetes operators running in the target cluster are responsible for pulling the secrets from the key vault and making them available either as Kubernetes secrets or secrets volume mounted to the pod.
+
+For secret rotation ideas, see [Secrets Rotation on Environment Variables and Mounted Secrets](#secrets-rotation-on-environment-variables-and-mounted-secrets)
+
+For how to authenticate private container registries with a Service principal see: [Authenticated Private Container Registry](#authenticated-private-container-registry)
+
+All the below options provide the following:
+
+- Secrets are not stored in the repository
+- Supports Prometheus metrics for observability
+- Supports sync with Kubernetes Secrets
+- Supports Linux and Windows containers
+- Provides enterprise-grade external secret management
+- Easily scalable with multi-cluster and larger teams
+- Both solutions support either Service Principal or Managed Identity for [authentication with the Key Vault](https://learn.microsoft.com/en-us/azure/key-vault/general/authentication).
+
+### [Azure Key Vault Provider for Secrets Store CSI Driver](https://github.com/Azure/secrets-store-csi-driver-provider-azure)
+
+Azure Key Vault Provider (AKVP) for Secrets Store CSI Driver is an azure vault provider for [Kubernetes secret store CSI Driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver) allows you to get secret content stored in Azure Key vault instance and use the Secrets Store CSI driver interface to mount them into Kubernetes pods. Mounts secrets/keys/certs to pod using a CSI Inline volume.
```

Comment:
Sentence is awkward. Maybe "...that allows you to..."? Not sure...

Reply:
I see your point, reworded